tests/drivers/uart/uart_async_api/boards/frdm_mcxn236.overlay

```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright 2024 NXP
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * To test this sample connect P4_2 to P4_3
+ */
+dut: &flexcomm2_lpuart2 {};
```

Comment:
please do not use lpuart2, as there is a arduino uart interface defined for P4_3 P4_2 which is also can be used for FC2, and this is the board design purpose